### PR TITLE
Add cql worker

### PR DESCRIPTION
**Pull request by rdingwell**
This PR brings an updated cql-worker back into the project.  The updated worker now supports executing a cql library as a whole and returning the results for all of the define statements.

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -5,10 +5,17 @@ function* simpleGenerator() {
     yield n++;
 }
 
-let simpleCounter = simpleGenerator();
+
+const simpleCounter = simpleGenerator();
+
+let currentId = 0;
+export function getCurrentId (){
```
Comment (mickohanlon23) [inline src/utils.js]: Where is this exported function used?

File: test/main.test.js [hunk 2]
```diff
@@ -446,6 +446,7 @@ describe('CQL expression tests', async function() {
 
   it('Should execute ten CQL expressions to evaulate ten applicability conditions (all true)', async function() {
     let resolver = simpleResolver('./test/fixtures/applicabilityConditionResources.json');
+    debugger
```
Comment (mickohanlon23) [inline test/main.test.js]: Should this be removed?

File: src/main.js [hunk 3]
```diff
@@ -1,32 +1,144 @@
-import { Worker as NodeWorker } from 'worker_threads';
-import { createRequire } from 'module';
+import { Worker as NodeWorker } from "worker_threads";
+import { createRequire } from "module";
+import { initialzieCqlWorker } from "cql-worker";
```
Comment (mickohanlon23) [inline src/main.js:3]: I noticed you're changing all of the single quotes to double quotes in the files that you're editting. Is this intentional, or done because of a linter you're using? If we are considering changing all single quotes to double quotes, I wonder if we should do it in a separate PR, since doing it automatically in just the files that you're editting makes the quote style inconsistent throughout the codebase and makes the diffs pretty messy to read.

File: src/main.js [hunk 4]
```diff
@@ -616,83 +741,49 @@ export async function processActions(actions, patientReference, resolver, aux, e
...
+    // if (Array.isArray(activityDefinition?.library)) {
+    //   const libRef = activityDefinition.library[0];
+
+    //   // Check aux for objects necessary for CQL execution
+    //   let elmJsonDependencies = aux.elmJsonDependencies ?? [];
+    //   const valueSetJson = aux.valueSetJson ?? {};
+    //   const cqlParameters = aux.cqlParameters ?? {};
+
+    //   const elmJsonKey = Object.keys(elmJsonDependencies).filter(e => libRef.includes(e))[0];
+    //   let elmJson = elmJsonDependencies[elmJsonKey];
```
Comment (mickohanlon23) [inline src/main.js]: Are we keeping this code here purposefully?

File: src/main.js [hunk 5]
```diff
@@ -535,7 +649,12 @@ export async function processActions(actions, patientReference, resolver, aux, e
       if (activityDefinition?.relatedArtifact) {
         for (const relatedArtifact of activityDefinition.relatedArtifact) {
           if (relatedArtifact.url !== undefined) {
-            payloads.push({ contentAttachment: { url: relatedArtifact.url, title: relatedArtifact.display }});
+            payloads.push({
```
Comment (mickohanlon23) [inline src/main.js]: The number of styling changes throughout this PR makes it difficult to understand what exactly is changing. Can we consider removing all of the styling changes from this PR and moving them to a separate PR dedicated only towards styling, if it's something we feel strongly about fixing?

File: src/main.js [hunk 6]
```diff
@@ -38,7 +150,12 @@ export async function applyAndMerge(planDefinition, patientReference=null, resol
  * @param {Object} aux - Auxiliary resources and services
  * @returns {Promise<Object[]>} Array of resources: CarePlan, RequestGroup, and otherResources
  */
-export async function applyPlan(planDefinition, patientReference=null, resolver=null, aux={}) {
+export async function applyPlan(
```
Comment (mickohanlon23) [inline src/main.js]: I tested this PR by upgrading the encender dependency in ccsm-cds-dashboard to reflect this branch.

I'm seeing the following error when I test:
```
main.js:13 Uncaught (in promise) TypeError: (0 , module__WEBPACK_IMPORTED_MODULE_1__.createRequire) is not a function
   at main.js:13:1
   at executeCQL (main.js:16:1)
   at applyPlan (main.js:156:1)
   at async applyCds (index.js:78:1)
```
This error is appearing when I test with both the local FHIR patients in the `tests-fhir` path, as well as when launching via SMART on FHIR using https://launch.smarthealthit.org/.

**Review by mickohanlon23 — CHANGES_REQUESTED** [review]
I left a few review comments. My main feedback is that I wasn't able to get the CDS Dashboard running with this version of encender, and the number of stylistic changes made it difficult to parse out exactly what was changing in this PR.

File: src/utils.js (hunk 1 above)
Comment (rdingwell) [reply src/utils.js]: That was already in the library, it's green because it got moved to a new location.

File: src/main.js (hunk 6 above)
Comment (rdingwell) [reply src/main.js]: This is the code that Mike and I have both been testing against.  But we have also been doing while having the dashboard pull in a branch of cql-execution.  That may very well be the issue, it's a reason I really don't really like using git urls for package dependencies

File: src/utils.js (hunk 1 above)
Comment (rdingwell) [reply src/utils.js]: Actually I was using that for debugging, it's getting removed.

File: src/main.js (hunk 6 above)
Comment (mickohanlon23) [reply src/main.js]: CQL execution still gets brought in by CQL Worker, right? I tested by:
- changing the `encender` dependency in package.json to `"github:ccsm-cds-tools/encender#add_cql_worker"`
- clearing my `node_modules` folder
- running `npm install`
- running `npm upgrade encender` to update to the latest version of this encender branch

When doing that, I was left with `cql-execution` version `3.0.1` in my `node_modules` folder, which according to `package-lock.json` was brought in through `cql-worker`.

File: src/main.js (hunk 3 above)
Comment (rdingwell) [reply src/main.js:3]: That was done by a linter.  I dont know if I can change everything that it changed.

File: src/main.js (hunk 6 above)
Comment (rdingwell) [reply src/main.js]: I just looked into this a bit.  I think it depends on what dashboard branch you may have tested against.  That is an error when the worker factory is not passed to encender.  That code may be commented out in your version of the dashboard so it needs to be uncommented and the worker factory passed into the aux variable

Comment (mickohanlon23) [reply src/main.js]: Good catch. I was using the `pilot` branch of [ccsm-cds-dashboard](https://github.com/ccsm-cds-tools/ccsm-cds-dashboard/tree/pilot). But in https://github.com/ccsm-cds-tools/ccsm-cds-dashboard/pull/58, we removed the `WorkerFactory`, since it wasn't used in the `cql-execution` branch of Encender that we were using.

I just created a new branch, [restore-worker-factory](https://github.com/ccsm-cds-tools/ccsm-cds-dashboard/tree/restore-worker-factory) that restores the commented out WorkerFactory code and updates encender to use this branch. Testing with that dashboard branch was successful for both the local FHIR patients and the SMART on FHIR patients! So once we merge this in and update the encender dependency in the dashboard, we'll just have to remember to add the WorkerFactory code back to the Dashboard.

File: src/main.js (hunk 5 above)
Comment (mickohanlon23) [reply src/main.js]: Formatting changes have been removed in https://github.com/ccsm-cds-tools/encender/pull/15/commits/891bfbdc776bd725b439903ead71b6f733ff423e.

File: src/main.js (hunk 3 above)
Comment (mickohanlon23) [reply src/main.js:3]: Formatting changes, including quote changes, have been removed in https://github.com/ccsm-cds-tools/encender/pull/15/commits/891bfbdc776bd725b439903ead71b6f733ff423e.

**Review by mickohanlon23 — APPROVED** [review]
All comments were resolved. I walked through the code changes with @rdingwell, and changes look good upon review. 

I tested this branch of encender using the `restore-worker-factory` branch of ccsm-cds-dashboard (https://github.com/ccsm-cds-tools/ccsm-cds-dashboard/tree/restore-worker-factory), and test patients are loading as expected. I still saw a boost in performance from running this branch, and the dashboard no longer freezes during CDS execution. This can be seen clearly by the spinner once again showing up when a toggle switch is flipped. Thanks @rdingwell!
